File: bonusQuestions/ThirdLightbulbs/fibonacci.java
package bonusQuestions.ThirdLightbulbs;

public class fibonacci {
    
    public static int fibonacciNum(int n) {
        int prevPrev = 0, prev = 1, sum = 0;
        if(n <= 1){
            return n;
        }
        else{
            for(int i = 2; i < n; i++){
                sum = prev + prevPrev;
                prevPrev = prev;
                prev = sum;
            }
        }
        return sum;

    }

    public static void main(String[] args) {
        int ans = fibonacciNum(7); //Fibonacci Sequence: 0,1,1,2,3,5,8,13,21
        System.out.println(ans);
    }
}

File: bonusQuestions/ThirdLightbulbs/fibonacci.py
def fibonacciNum(n):
    prevPrev, prev, sum = 0,1,0
    if n <= 1: return n
    else:
        for i in range(2,n):
            sum = prev + prevPrev
            prevPrev = prev
            prev = sum
    return sum

ans = fibonacciNum(7)
print(ans)